libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigatorImpl.kt

```diff
@@ -154,12 +157,18 @@ object MapboxNativeNavigatorImpl : MapboxNativeNavigator {
 
     // Routing
 
-    override suspend fun setPrimaryRoute(
-        routeWithStartingLeg: Pair<NavigationRoute, Int>?
-    ): RouteInfo? = suspendCancellableCoroutine { continuation ->
-        navigator!!.setPrimaryRoute(
-            routeWithStartingLeg?.first?.nativeRoute(),
-            routeWithStartingLeg?.second ?: 0
+    override suspend fun setRoutes(
+        primaryRoute: NavigationRoute?,
+        startingLeg: Int,
+        alternatives: List<NavigationRoute>,
+    ): Expected<String, SetRoutesResult> = suspendCancellableCoroutine { continuation ->
+        navigator!!.setRoutes(
```

Comment:
@DenisPryt , @SiarheiFedartsou , what is the sate of NN if `setRoutes` returns error? Is it the same as before setting a route?
cc @LukasPaczos , @RingerJK

Reply:
I'm asking because I don't know what the Android SDK should do in case of error. Keep the same state as before setting route?

Reply:
Tracking the handling of errors in https://github.com/mapbox/mapbox-navigation-android/issues/5686, let's leave this as-is in this PR.